test/upgrade/prober/wathola/sender/services_test.go

```diff
@@ -74,20 +75,19 @@ func TestUnsupportedEventSender(t *testing.T) {
 }
 
 func waitForPort(t *testing.T, port int) {
-	i := 0
-	for {
-		conn, err := net.Dial("tcp", fmt.Sprintf(":%d", port))
-		if conn != nil {
-			assert.NoError(t, conn.Close())
-			return
-		}
-		if err != nil {
-			time.Sleep(time.Millisecond)
-			i++
-		}
-		if i > 10_000 {
-			t.Fatal("timeout waiting for port to open: ", port)
+	if err := wait.PollImmediate(time.Millisecond, 10*time.Second, func() (bool, error) {
+		conn, conErr := net.Dial("tcp", fmt.Sprintf(":%d", port))
+		defer func() {
```

Comment:
If you return just `true, conn.Close()` the outer `err != nil` condition will catch any error and the Fatalf will properly log the error (as I wrote in the suggestion earlier). It would also remove the need for the defer (which I don't understand here)
What is the goal of `defer` here? It looks redundant.

Reply:
`defer` here is just defensive programming. It will work well even in case if there will be `conn != nil && conErr != nil` in the same time

Reply:
Isn't `net.Dial` already ensuring that when `conErr == nil` the `conn` will always be non-nil? Maybe we're too defensive here. But no big deal.